src/libbson/src/bson/bson-decimal128.c

```diff
@@ -501,7 +501,7 @@ bson_decimal128_from_string_w_len (const char *string,     /* IN */
    size_t first_digit = 0;           /* The index of the first non-zero digit */
    size_t last_digit = 0;            /* The index of the last digit */
 
-   int32_t exponent = 0;
+   int64_t exponent = 0;
```

Comment:
```suggestion
   int32_t exponent = 0;
```

Change can be reverted since `exponent` is assigned after casting to `int32_t`.